src/event-bridge/event-bus.ts

```diff
@@ -83,7 +113,109 @@ abstract class EventBusBase<E extends EventBusRuleInput>
   public static readonly FunctionlessType = "EventBus";
   readonly functionlessKind = "EventBus";
 
-  constructor(readonly bus: aws_events.IEventBus) {}
+  readonly bus: aws_events.IEventBus;
+
+  constructor(bus: aws_events.IEventBus) {
+    this.bus = bus;
+    return makeCallable(this, (call: CallExpr, context: CallContext) => {
+      if (isASL(context)) {
+        this.bus.grantPutEventsTo(context.role);
+
+        // Lets validate and normalize that the events are
+        const eventObjs = call.args.reduce(
+          (events: ObjectLiteralExpr[], arg) => {
+            if (isArrayLiteralExpr(arg.expr)) {
+              if (
+                !arg.expr.items.every((item): item is ObjectLiteralExpr =>
+                  isObjectLiteralExpr(item)
+                )
+              ) {
+                throw Error(
+                  "Event Bus put events must use inline object parameters. Variable references are not supported currently."
+                );
+              }
```

Comment:
Make sure prettier is running

Reply:
Indentation looked weird

Reply:
Looks like its not working on mobile, web looks fine

![image](https://user-images.githubusercontent.com/289213/162794154-ac7c0d0f-b0f2-4bcb-9541-46028eef6762.png)